src/cart/crt_corpc.c

```diff
@@ -121,6 +122,10 @@ crt_corpc_initiate(struct crt_rpc_priv *rpc_priv)
 		}
 	}
 
+	/* Inherit a timeout from a source */
+	src_timeout = rpc_priv->crp_req_hdr.cch_src_timeout;
```

Comment:
do you really need src_timeout?
can you just 
rpc_priv->crp_timeout_sec = rpc_priv->crp_req_hdr.cch_src_timeout; ?

Reply:
Currently we don't but in subsequent patches when we switch to a deadline model it will get adjusted further here so that each subsequent hop timeout doesnt exceed overall